### DISCUSSION TITLE
Listening to specific updates on StateNotifier from other stateNotifiers and/or Providers

### DISCUSSION THREAD

**Discussion by mihael998** [rrousselGit/riverpod#1200] [Q&A] [answered]
Hi, I have been struggling with something for a while.
Here's what I'm trying to accomplish:

In my application I have different providers, simple ones, and also StateNotifier ones.
A lot of my StateNotifiers manage operations on lists (like add, remove, update) and so on.
In my app I have two State Notifiers with their providers like so:

```
class FirstNotifier
    extends StateNotifier<FirstState> {

  ArchitectureCanvasNotifier( FirstState state)
      : super(state);

  updateStateInformation(){
    state=...
  }
}

final firstNotifierProvider= StateNotifierProvider<FirstNotifier, FirstState>(
  (ref) {
    final notifier = FirstNotifier(
      FirstState(),
    );

    return notifier;
  },
);
```

```
class SecondNotifier
    extends StateNotifier<List<Model>> {

  ArchitectureCanvasNotifier( List<Model> state)
      : super(state);

  removeElement(){
    state=[...state.value!]..removeWhere(...)
  }
}

final secondNotifierProvider= StateNotifierProvider<SecondNotifier, List<Model>>(
  (ref) {
    final notifier = SecondNotifier(
      [],
    );

    return notifier;
  },
);

```

What I'm trying to accomplish is to update the FirstNotifier whenever `removeElement` is called on the second notifier.

To do so what are my options?
I could call `updateStateInformation` inside `removeElement` by passing a reference of `ref` to the SecondNotifier. From what I have understood there is nothing wrong in passing the ref object.
What I'm trying to reach though is an "Event" like process. I would like to put the logic of "listening" to `removeElement` in the FirstNotifier or FirstNotifierProvider.
So something like:
```
final firstNotifierProvider= StateNotifierProvider<FirstNotifier, FirstState>(
  (ref) {
    final notifier = FirstNotifier(
      FirstState(),
    );

    ref.listen<List<Model>>(secondNotifierProvider,(old,new){
      
    });
    return notifier;
  },
);

```

The problem is this: how can I listen to specific changes? Inside the listen callback I only have a reference to the new List<Model>. Do I have to "compute" manually the changes based on old and new value or there is a way to know that `removeElement` has been called?

**Comment by rrousselGit** [comment, marked as answer]
Your reasoning is correct.

What you're looking for (an event mechanism for every StateNotifier method) does not exist. Placing the logic within the `removeElement` method by passing ref like you've mentioned would be the correct solution

**Reply by mihael998** [reply]
🥲😭
Thank you for the quick response!
By using some kind of reactive pattern (for example stream with streamProvider) would it be possible, or better would it be a "recommended" way to do it? (I could look into it but I suppose you may have a "ready answer" based on your experience in discussing RiverPod things 😁)
